### PR TITLE
Implement `AppUserAuth` for Github App user tokens

### PR DESCRIPTION
**Pull request by EnricoMi**
Allows to refresh Github App user token. Integrates `ApplicationOAuth` into `github.Auth`.

### CODE REVIEW

**Comment by EnricoMi** [comment]
CC @rigaspapas

File: github/AccessToken.pyi
Comment (JLLeitschuh) [inline github/AccessToken.pyi:1]: Given this file is so small, you could likely delete it and just put the types on the implementation

**Review by JLLeitschuh — APPROVED** [review]
Not sure if I can review this on substance, but looks good to me!

File: github/AccessToken.pyi
Comment (EnricoMi) [reply github/AccessToken.pyi:1]: Good point! I started this, but this unveiled lots of typing warnings, discussed and to be fixed here: https://github.com/PyGithub/PyGithub/pull/2463#discussion_r1227714353

**Pull request merged**